internal/cmd/tls.go

```diff
@@ -0,0 +1,255 @@
+				return err
+			}
+
+			cert, err := os.ReadFile(certFilepath)
+			if err != nil {
+				return err
+			}
+
+			key, err := os.ReadFile(keyFilepath)
+			if err != nil {
+				return err
+			}
+
+			req := coder.UploadCertificateRequest{
+				Certificate: cert,
+				PrivateKey:  key,
+			}
+
+			resp, err := client.UploadCustomCertificate(ctx, req)
+			if err != nil {
+				return err
+			}
+
+			if !resp.Success {
+				return xerrors.New("Failed to apply custom TLS certificate")
+			}
+
+			clog.LogSuccess("Uploaded custom TLS certificate")
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&certFilepath, "cert", "c", "", "Full path to a PEM encoded certificate file")
+	cmd.Flags().StringVarP(&keyFilepath, "key", "k", "", "Full path to a PEM encoded private key file")
+	return cmd
+}
+
+func tlsACME() *cobra.Command {
+	var (
+		showProviderInfo bool
+
+		agreeTOS    bool
+		email       string
+		domains     []string
+		dnsProvider string
+		credentials map[string]string
+	)
+	cmd := &cobra.Command{
+		Use:   "acme",
+		Short: "Generate certificate via Let's Encrypt",
+		Example: `
+tls acme --info
+tls acme --email me@example.com --domains a.example.com --domains b.example.com --provider route53 --credentials AWS_ACCESS_KEY_ID=your-key-id --credentials AWS_SECRET_ACCESS_KEY=your-secret-key --credentials AWS_REGION=your-region`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx, true)
+			if err != nil {
+				return err
+			}
+
+			if showProviderInfo {
+				providers, err := client.GetLetsEncryptProviders(ctx)
+				if err != nil {
+					return err
+				}
+				if err = tablewriter.WriteTable(cmd.OutOrStdout(), len(providers.Providers), func(i int) interface{} {
+					return coderutil.ProviderInfosTable(providers.Providers[i])
+				}); err != nil {
+					return err
+				}
+				return nil
+			}
+
+			catcher := grip.NewBasicCatcher()
+			if !agreeTOS {
+				catcher.Add(ErrMissingRequiredParameter{"agree-tos"})
+			}
+			if email == "" {
+				catcher.Add(ErrMissingRequiredParameter{"email"})
+			}
+			if len(domains) == 0 {
+				catcher.Add(ErrMissingRequiredParameter{"domains"})
+			}
+			if dnsProvider == "" {
+				catcher.Add(ErrMissingRequiredParameter{"provider"})
+			}
+			if catcher.HasErrors() {
+				return catcher.Resolve()
+			}
+
+			req := coder.GenerateLetsEncryptCertRequest{
+				Email:       email,
+				Domains:     domains,
+				DNSProvider: dnsProvider,
+				Credentials: credentials,
+			}
+
+			resp, err := client.GenerateLetsEncryptCert(ctx, req)
+			if err != nil {
+				return err
+			}
+
+			if !resp.Success {
+				return xerrors.New("Failed to generate certificate via Let's Encrypt")
+			}
+
+			clog.LogSuccess("Generating certificate via Let's Encrypt. This may take a few minutes to complete.")
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVarP(&showProviderInfo, "info", "i", false, "Show supported DNS providers and required credentials for each")
```

Comment:
I would suggest avoiding the short forms (`i`) as it makes things easier to mix up, and we can always add short forms later, but cannot remove them without breaking backward compatibility